Run stress tests out of Jasmine

bigmontz opened the pull request:
The goal of this change is improve the stability of the stress test avoiding unexpected timeouts.

bigmontz committed:
Run stress tests out of Jasmine

The goal of this change is improve the stability of the stress test avoiding unexpected timeouts.

2hdddg commented on test/stress-test.js:
```diff
@@ -0,0 +1,743 @@
+  )
+}
+
+function assertAllAddressesServedReadQueries (addresses, readQueriesByServer) {
```
I'm don't believe that is safe to do this check if the cluster is reforming while the tests are running we might end up with new members of the cluster right before this check.
Load balancing algorithm is better tested with other means..

bigmontz replied:
Sure, I will remove all the `verifyCausalClusterMembers` function, since it's could cause problem.

2hdddg commented on test/stress-test.js:
```diff
@@ -0,0 +1,743 @@
+  })
+}
+
+function assertAllAddressesServedSimilarAmountOfReadQueries (
```
See above

2hdddg commented on test/stress-test.js:
```diff
@@ -0,0 +1,743 @@
+import sharedNeo4j from './internal/shared-neo4j'
+import { resolve } from 'uri-js'
+
+const TEST_MODES = {
```
Would be nice to be able to configure this from testkit invocation.
I'm not fully convinced that specifying the number of commands is the best way to run this in stress tests but  it will work in drop pipeline.

From testkit it would be easier to specify the number of seconds that the suite should be running. The scenario is: 
* Start the cluster
* Start stress test suite
* Wait x seconds
* Add another member to cluster
* Wait y seconds
* Remove a member from the cluster
* Wait z seconds

In this case we need to know that the test runs for a bit longer than x+y+z seconds otherwise testkit should fail the test since it cannot verify that the driver could handle rolling cluster members.

bigmontz replied:
Yes, I will do this one today in a second iteration, this PR was more about porting the existing code.

bigmontz replied:
Solved in this PR

2hdddg commented on test/stress-test.js:
```diff
@@ -0,0 +1,743 @@
+  fastest: {
+    commandsCount: 10000,
+    parallelism: 24,
+    maxRunTimeMs: 3 * 60000 // 3 minutes
```
Is maxRunTimeMs used?

bigmontz replied:
Not anymore, it should be removed.

bigmontz committed:
addressing comments

bigmontz committed:
stress-test: Make it able to run for x seconds

This feature is important because it enables testkit to perform stress tests replacing
parts of the cluster.
